test/functional/product/create_test.rb

```diff
@@ -0,0 +1,38 @@
+require File.join(File.dirname(__FILE__), '../test_helper')
+require File.join(File.dirname(__FILE__), '../organization/organization_helpers')
+require File.join(File.dirname(__FILE__), '../sync_plan/sync_plan_helpers')
+require File.join(File.dirname(__FILE__), '../product/product_helpers')
+
+describe "create a product" do
+  include OrganizationHelpers
+  include SyncPlanHelpers
+  include ProductHelpers
+
+  it 'by organization ID and product name' do
+    api_expects(:products, :create, 'create a product') do |par|
+      par['organization_id'] == 1 && par['name'] = 'product1'
+    end
+
+    assert(0, run_cmd(%w(product create --organization-id 1 --name product1)).exit_code)
```

Comment:
I don't think this is working like you think: http://ruby-doc.org/stdlib-2.0.0/libdoc/minitest/rdoc/MiniTest/Assertions.html#method-i-assert

You probably wanted `assert_equal`.